tools/test/h5dump/h5dumpgentest.c

```diff
@@ -14289,6 +14338,10 @@ gent_test_reference_external(void)
     if (H5Dwrite(dataset, H5T_STD_REF, H5S_ALL, H5S_ALL, H5P_DEFAULT, ref_wbuf) < 0)
         return 1;
 
+    /* Destroy references to free allocated memory */
+    for (i = 0; i < SPACE1_DIM1; i++)
+        H5Rdestroy(&ref_wbuf[i]);
```

Comment:
However, I could see this missing destroy call being a source of problems. Have you verified whether this change alone fixes the crash seen?